packages/wallets/src/evm/connectors/smart-wallet/lib/account.ts

```diff
@@ -143,4 +141,49 @@ export class AccountAPI extends BaseAccountAPI {
   async isAcountDeployed() {
     return !(await this.checkAccountPhantom());
   }
+
+  async isAccountApproved() {
+    if (!this.params.erc20PaymasterAddress || !this.params.erc20TokenAddress) {
+      return true;
+    }
+
+    const swAddress = await this.getCounterFactualAddress();
+    const erc20Token = await this.sdk.getContract(
+      this.params.erc20TokenAddress,
```

Comment:
might be worth pass the 'allowance' abi here (we actually have the IERC20 abi that you can import from here) to make sure we don't fail on erc20 contract we cant resolve

Reply:
yeah definitely would make it a lot faster good call